File: scripts/test_open_case.py
from playwright.sync_api import sync_playwright

with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)
    page = browser.new_page(viewport={"width": 1280, "height": 800})
    
    console_logs = []
    page.on("console", lambda msg: console_logs.append(f"[{msg.type}] {msg.text}"))
    
    try:
        page.goto('http://localhost:5173')
        page.wait_for_load_state('networkidle')
        
        # Click on Cassino (where DiscountRoulette is)
        page.locator('text=Cassino').first.click(timeout=3000)
        page.wait_for_timeout(1000)
        
        # Click on the first case
        case = page.locator('.group.relative.bg-zinc-900\\/50').first
        case.click(timeout=3000)
        page.wait_for_timeout(1000)
        
        # Click "Abrir 1x"
        open_btn = page.locator('text=Abrir 1x')
        open_btn.click(timeout=3000)
        
        # Wait for spin to finish (around 10 seconds)
        print("Esperando o giro (10s)...")
        page.wait_for_timeout(10000)
        
        # Print logs
        print("\n--- CONSOLE LOGS ---")
        for log in console_logs:
            print(log)
            
    except Exception as e:
        print(f"Erro no teste: {e}")
        
    browser.close()

File: test_open_case.mjs
import { chromium } from 'playwright';

(async () => {
  console.log('Iniciando navegador...');
  const browser = await chromium.launch({ headless: true });
  const page = await browser.newPage({ viewport: { width: 1280, height: 800 } });
  
  const consoleLogs = [];
  page.on('console', msg => {
    consoleLogs.push(`[${msg.type()}] ${msg.text()}`);
    console.log(`[BROWSER] ${msg.type()}: ${msg.text()}`);
  });
  
  try {
    console.log('Navegando para o site...');
    await page.goto('http://localhost:5173', { waitUntil: 'networkidle' });
    
    // Tenta clicar em "Entrar" se não estiver logado
    // Wait, playwright is anonymous, it won't be logged in!
    // I need to login first or the insert WILL fail.
    
    // Check if logged in
    const isLoginButton = await page.locator('text=Entrar').isVisible();
    if (isLoginButton) {
      console.log('Site requer login. Não podemos testar inserir sem logar.');
    } else {
      console.log('Usuário já parece estar logado?');
    }
  } catch (e) {
    console.error('Erro:', e);
  } finally {
    await browser.close();
  }
})();
